design-proposals/shadow-node.md

```diff
@@ -0,0 +1,286 @@
+- patches built-in labels with `feature.node.kubernetes.io/` prefix. See list [here](https://kubernetes-sigs.github.io/node-feature-discovery/v0.15/usage/features.html#built-in-labels).
+  - These labels are similar but do not exactly match the ones Kubevirt is deploying right now (See [comparison](https://www.diffchecker.com/qA2EvefO/) of local dev node deployed using KubevirtCI)
+  - This NFD feature will not be used for this solution.
+
+Example:
+Instead of directly patching these labels to the node: `cpu-feature.node.kubevirt.io/abm: "true"`, `cpu-feature.node.kubevirt.io/adx: "true"`, it would deploy the following CR:
+```yaml
+apiVersion: nfd.k8s-sigs.io/v1alpha1
+kind: NodeFeature
+metadata:
+  labels:
+    nfd.node.kubernetes.io/node-name: node-1
+  name: node-1-labeller-sync
+spec:
+  labels:
+    cpu-feature.node.kubevirt.io/abm: "true"
+    cpu-feature.node.kubevirt.io/adx: "true"
+#...
+```
+See full CR [here](https://paste.centos.org/view/ef218827)
+
+Pros:
+- No need to invent the wheel.
+- Re-examine current methods to make sure if we still need them is a good practice. In this case we may find out that patching is no longer needed, and in the long run need to maintain less code.
+- This steps can be broken to multiple seemingly unrelated PRs, that are easier to review.
+
+Cons:
+- Relying on a third part tools.
+
+## Solution option #4: [Ruled-out] Using status section
+Same proxy solution as primary solution where virt-hadler updates the labels/annotations somewhere, and it gets copied to the node by virt-controller, but instead of using a new CRD as the medium - writing the labels/Annotations to the status section of an existing CRD like kubevirt CR.
+
+Pros:
+- Avoid introducing a new CRD to the project.
+
+Cons:
+- [Deal-breaker] The status section is considered a user facing API section. Dynamically adding/updating 150+ technical labels there may only confuse the user and thus is not advisable.
+
+## Solution option #5: [Ruled-out] Using the virt-handler pod as a medium to pass the labels/annotations
+Same proxy solution as primary solution where virt-handler updates the labels/annotations somewhere, and it gets copied to the node by virt-controller, but instead of using a new CRD as the medium - writing the labels/annotations to the virt-handler pods's labels/annotations.
+Labels/Annotations that need to pass to the node should get a special prefix (like "nodesync/"), so that regular labels don't get passed to the node by mistake.
+
+Pros:
+- Avoid introducing a new CRD to the project.
+- A simple and clean approach.
+
+Cons:
+- [Deal-breaker] It requires giving the virt-handler RBAC permission to update pods, which is generally considered risky.
+- It's only relevant for passing labels/annotations. CRD approach allows for future expansions.
+
+## Solution option #6: [Ruled-out] Fine-tune each virt-handler RBAC to only patch its own node.
```

Comment:
I agree. I mentioned this in the follow up steps section in the end of the proposal.

BTW I saw several interesting approaches to this that are worth looking into when it comes to the table: 
1. Using validatingadmissionpolicy https://github.com/openshift/origin/pull/28673
2. Using validating webhook as I think you also suggested IIRC (https://github.com/ovn-org/ovn-kubernetes/blob/master/docs/node-identity.md)

Reply:
Yep 1 is a something to check out. 2 is something we have to avoid on the node object itself. As I mentioned it would put kubevirt in a critical cluster health path (but we can have either 1 or 2 on our shadow node cr, since we own the shadow node cr and we don't influence cluster health there with our webhook availability)